### DISCUSSION TITLE
How can I add another programming language support

### DISCUSSION THREAD

**Discussion by Killpit** [github/codeql#17405] [Q&A] [answered]
How can I add Rust or Flutter/Dart support if I want to add them

**Comment by mbg** [comment, marked as answer]
Hi @Killpit 👋🏻 

Thanks for the question and wanting to add support for additional programming languages. While this is possible, we don't have any documentation or guides for how to do so. One of my colleagues has a repository with a minimal implementation needed to support a language at https://github.com/aibaars/codeql-kaleidoscope which may be a good starting point to look into to give you an idea of how you can create a custom extractor and how that can be used with the standard CodeQL CLI commands.

**Reply by Killpit** [reply]
Cheers man

**Reply by Killpit** [reply]
Also I asked him about the project and how I can add language support in Rust. Couldn't look further but I think it's specific for Kaleidoscope and I might have to create it on official CodeQL repo (on a fork) to add Rust and Flutter/Dart support I think

**Reply by mbg** [reply]
Yes, my colleague's repo contains the minimum necessary to support a small toy language (Kaleidoscope) for CodeQL. Like I said, it's just an example where you can look at the code to see what's involved in supporting a language.

I don't believe that you need to necessarily work on a clone of this repository.

**Reply by intrigus-lgtm** [reply]
By the way, official Rust support is being added:
https://github.com/github/codeql/pulls?q=is%3Apr+is%3Aopen+Rust%3A

**Reply by Killpit** [reply]
What about Flutter/Dart and how can I involve with the Rust support @intrigus-lgtm

**Reply by Killpit** [reply]
One thing I'd like to here, how can we add language support project such as `flutter create` as an example, how can we start with all the possible tools on the repo

**Reply by icepacket** [reply]
Hi @mbg, any chances you could get [it](https://github.com/aibaars/codeql-kaleidoscope) running... I tried following the instructions but ended up with errors while running `scripts/create-extractor-pack.sh`

**Reply by mbg** [reply]
@icepacket you should probably open an issue on that repo, if you have a problem getting it to work. You'll also have to include more details about what errors you run into if you want someone to be able to help. That said, that repo is just an unofficial example of the sort of thing you may have to do and isn't an official piece of documentation or something we otherwise support.

**Reply by icepacket** [reply]
@mbg I created a issue but haven't heard back yet. And, I am clueless about getting started with this. Do you have any suggestions which might help in adding support for a new programming language support? The final option for me is to read hundreds of files from existing supported languages.
